lib/AST/Type.cpp

```diff
@@ -3214,6 +3214,38 @@ Type Type::substDependentTypesWithErrorTypes() const {
                     SubstFlags::UseErrorType));
 }
 
+Type Type::substTypeParametersWithUnresolved() const {
```

Comment:
I don't want to see these become methods on Type since they're hacks meant to paper over other problems in CSDiag, and not generally useful.